### ISSUE TITLE
Support for HTTP Response Status

### ISSUE THREAD

**Issue by rizwan-0** [absinthe-graphql/absinthe_plug#272]
Hello everyone, according to the function in [Absinthe.Plug](https://github.com/absinthe-graphql/absinthe_plug/blob/35b0661b24474cec1b857e8d2cc04f01077026d3/lib/absinthe/plug.ex#L287) , there can be only `200 | 400 | 405 | 500` error codes to return, as also defined here in [encode()](https://github.com/absinthe-graphql/absinthe_plug/blob/35b0661b24474cec1b857e8d2cc04f01077026d3/lib/absinthe/plug.ex#L591).

Is there a specific reason this library doesn't have any option for all / custom status codes.
I was thinking of something like this:
```
def call(conn, config) do
    config = update_config(conn, config)
    {conn, result} = conn |> execute(config)

    result
    |> case do
      {:input_error, msg} ->
        conn
        |> encode(400, error_result(msg), config)

      {:ok, %{"subscribed" => topic}} ->
        conn
        |> subscribe(topic, config)

      {:ok, %{errors: [ %{code: code} |_]} = result} ->
        conn
        |> encode(code, result, config)

      {:ok, %{errors: _} = result} ->
        conn
        # 406 = not_acceptable
        |> encode(406, result, config)

      {:ok, %{data: _} = result} ->
        conn
        |> encode(200, result, config)

      {:ok, result} when is_list(result) ->
        conn
        |> encode(200, result, config)

      {:error, {:http_method, text}, _} ->
        conn
        |> encode(405, error_result(text), config)

      {:error, error, _} when is_binary(error) ->
        conn
        |> encode(500, error_result(error), config)
    end
  end
```

Here, I am getting only the first code value in a list, as in my cases there is only one error entry in list but the point is to have some custom defined or user preferred return codes so the error handling on client is done easily.

Any help or alternative is welcome for this.

**Comment by epinault** [comment]
what about 429? this would be a good one to have because when logic is in the resolver due to business logic, I don t seem to have a way to override that  and I would think the spec support that?

**Comment by epinault** [comment]
@benwilson512 what is your take on this?

**Comment by epinault** [comment]
I see, I added to the issue because tthat how I found the status code list suppported in the code and that someone else had a similar issue and was curious about it. but I can use the forum next time then

can you point me to the spec where you see those code ? cause I am not seeing anything in. https://spec.graphql.org/October2021/#sec-Response-Format

**Comment by benwilson512** [comment]
Can you elaborate? What code does Absinthe return that is invalid?

**Comment by epinault** [comment]
ah gotcha part of my confusion then. thanks for looking into it. The problem with the before_send is that it run the logic first and is done after absinthe did the work right? so the throttling need to happen before

**Comment by benwilson512** [comment]
Yes that's correct. Throttling should happen in the plug layer before Absinthe executes at all, and I think it'd be perfectly valid to return a 429 in such cases. That'd just happen earlier in the stack and so `Absinthe.Plug` wouldn't execute at all. I'd consider such throttling and its associated response part of this clause:

> Note: A status code in the 4xx or 5xx ranges or status code 203 (and maybe others) could originate from intermediary servers;

It's not really a "server" but it's intermediary request handling middleware.

**Comment by epinault** [comment]
Ok. Make sense ! Thank you !  Do you know if there is a way to just parse the query / mutation out of the body  with absinthe? ( prior to hit absinthe plug ) .
